### PR TITLE
Living Mannequin + Cult Mannequins

### PR DESCRIPTION
**Pull request by JMWTurner**
https://dl.dropboxusercontent.com/u/33846352/dreamseeker%202016-10-27%2021-08-19-72.webm

https://dl.dropboxusercontent.com/u/33846352/dreamseeker%202016-10-28%2001-12-41-00.webm

Mappers and vault designers are gonna love those.

:cl:
- rscadd: When using a staff of animation on an already animated object, the object will see the firer as its new master, allowing wizards to take their ownership from other wizards for instance.
- rscadd: Added a new mob, the Living Mannequin. Mannequins will awaken when hit by a bolt of animation. Additionally, I've added a few variables that allow mappers to create mannequins that automatically awaken when they spawn, or when a mob walks next to them, or when you try to strip them, and that make nearby mannequins awaken as well. Destroying a Living Mannequin drop their items. If the mannequin was holding a highly damaging item when it woke up, it'll fight using that item. It doesn't have the dexterity to fire guns however.
- rscadd: Added cult mannequins. They appear when Nar-Sie cultifies a mannequin.

### CODE REVIEW

**Comment by jakmak3** [comment]
do cult maniquens have any other special properties?

**Comment by JMWTurner** [comment]
They don't attack cultists (well they do in the webm but I fixed that) they have barely more health than marble ones. By default they are also strip-trapped (if you try to strip them/wrench them/place items on them, they will awaken) and will awaken any mannequin in a 3 tiles radius when they awaken themselves. And they have kickass pedestals.

**Comment by jakmak3** [comment]
When influenced by the power of nar nar do they automatically come alive and begin trying to kill non cults? and is there a way to animate as a cult without summoned narsie/SoC?

**Comment by JMWTurner** [comment]
They come alive only if you try to strip them/wrench them/you hit them. They currently can only exist post-narsie, but narsie himself only changes their appearance, he doesn't cause them to come alive immediately.

File: code/modules/mob/living/simple_animal/hostile/mannequin.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	icon_state = "mannequin_wooden_human"
+	icon_living = "mannequin_wooden_human"
+	harm_intent_damage = 5
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	maxHealth = 30
+	health = 30
+
+/mob/living/simple_animal/hostile/mannequin/wood/breakDown()
+	visible_message("<span class='warning'><b>[src]</b> collapses!</span>")
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/mannequin.dm:67]: Won't you want it to inherit the parent proc?

File: code/game/objects/structures/mannequin.dm [hunk 2]
```diff
@@ -678,7 +708,47 @@
...
+	newPedestal.dir = dir
+	var/mob/living/simple_animal/hostile/mannequin/livingMannequin = new trueForm(loc)
+	livingMannequin.name = name
+	livingMannequin.icon_state = icon_state
+	livingMannequin.health = health
+	livingMannequin.maxHealth = maxHealth
+	livingMannequin.dir = dir
+	var/image/I = image('icons/effects/32x32.dmi',"blank")
+	I.overlays |= overlays
+	I.pixel_x = -1*clothing_offset_x
```
Comment (PJB3005) [inline code/game/objects/structures/mannequin.dm]: Just do `-clothing_offset_x`?

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by JMWTurner** [comment]
revision 1:
- added the wood and cyber living mannequin's custom breakdown procs.

File: code/game/objects/structures/mannequin.dm (hunk 2 above)
Comment (JMWTurner) [reply code/game/objects/structures/mannequin.dm]: done

**Comment by PJB3005** [comment]
Did you forget to push the commit?

**Comment by JMWTurner** [comment]
@PJB3005 as a matter of fact, I did

File: code/game/objects/structures/mannequin.dm (hunk 2 above)
Comment (JMWTurner) [reply code/game/objects/structures/mannequin.dm]: done for real

File: code/modules/mob/living/simple_animal/hostile/mannequin.dm (hunk 1 above)
Comment (JMWTurner) [reply code/modules/mob/living/simple_animal/hostile/mannequin.dm:67]: no, the point of this proc is that something custom for each subtype can be done on death.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**